### DISCUSSION TITLE
Possibility to show commit titles when reviewing a file in a PR

### DISCUSSION THREAD

**Discussion by matthijs166** [community/community#31585] [Pull Requests]
Hey everyone,

I'm reviewing a lot of pull requests lately and I'm wondering if the following is possible.

When I review the files of a pull request I am wondering sometimes why a certain choice is made by the developer.
This choice is often described inside the commit title.

It's very hard at the moment to review a pull request with a lot of small changes and hard to figure out the intent behind a change.

**Comment by jfullstackdev** [comment]
it's actually on the committer, my practice is to actually create a pull request with one intention or related intentions

if that is jumbled, yup, that's very hard to see

**Comment by lrynek** [comment]
@matthijs166
1️⃣ I use for that going commit by commit diff at the commits streamline at the bottom of the pull request and later to the `Files changed` tab to build up all the particular decisions to the final result.
2️⃣ It's also good to use an IDE or any other editor that supports git blame annotations for the lines at hand, so once switching to the pull request's branch locally, you can reference the actual code down to the respective commit where it was introduced.
